modules/administration/pages/auth-methods-sso.adoc

```diff
@@ -0,0 +1,209 @@
+You can install Keycloak directly on your machine, or run it in a container.
+In this example, we run Keycloak in a Docker container.
+For more information about installing Keycloak, see the Keycloak documentation at https://www.keycloak.org/getting-started/getting-started-docker.
+
+
+
+.Procedure: Setting Up the Identity Service Provider
+. Install KeyCloak in a Docker container, according to the Keycloak documentation.
+. Run the container using the ``-td`` argument to ensure the process remains running:
++
+----
+docker run -td --name=idp -p 8080:8080 -e KEYCLOAK_USER=<user> -e KEYCLOAK_PASSWORD=<password> quay.io/keycloak/keycloak:9.0.2
+----
+. Sign in as a privileged user, and use the Keycloak {webui} to create a realm using these details:
++
+* In the ``Name`` field, enter a name for the realm.
+For example, ``SUMA``.
+* Toggle the ``Enabled`` switch to ``On``.
+* In the ``Endpoints`` field, type ``SAML 2.0 Identity Provider Metadata``.
+This endpoint is ``<IdP_Address>/auth/realms/<Realm_Name>/protocol/saml/descriptor`` which describes the endpoints and certificate in the {productname} configuration file.
+// Probably needs more explanation here. --LKB 20200415
+
+
+
+When you have Keycloak running and set up, you can add the endpoints.
+Keycloak refers to endpoints as clients.
+
+
+
+.Procedure: Adding Endpoints as Clients
+. In the Keycloak {webui}, create a new client using these details:
++
+* In the ``Client ID`` field, enter the endpoint specified in the server configuration file as ``java.sso.onelogin.saml2.idp.entityid``.
+For example, ``https://<FQDN>/rhn/manager/sso/metadata``.
+* In the ``Client Protocol`` field, select ``SAML``.
+* Toggle the ``Includew AuthnStatement`` switch to ``On``.
+* Toggle the ``Sign Assertions`` switch to ``On``.
+* In the ``Signature Algorithm`` field, select ``RSA_SHA1``.
+* In the ``SAML Signature Key Name`` field, select the key.
+* In the ``Canonicalization Method`` field, select ``Exclusive``.
+. In the ``Fine Grain SAML Endpoint Configuration`` section, add the two endpoints using these details:
++
+* In both the ``Assertion Consumer Service`` fields, enter the endpoint specified in the server configuration file as ``java.sso.onelogin.saml2.sp.assertion_consumer_service.url``.
+For example, ``https://<FQDN>/rhn/manager/sso/acs``.
+* In both the ``Logout Service`` fields, enter the endpoint specified in the server configuration file as ``java.sso.onelogin.saml2.sp.single_logout_service.url``.
+For example, ``https://<FQDN>/rhn/manager/sso/sls``.
+
+
+
+When you have added the endpoints as clients, you can configure the client scope, and map the users between Keycloak and {productname}.
+
+
+
+.Procedure: Configuring Client Scope and Mappers
+. In the Keycloak {webui}, navigate to the menu:Clients[Client Scopes] tab and assign ``role_list`` as the default client scope.
+. Navigate to the menu:Clients[Mappers] tab and add a user attribute ``Uid`` mapper, using the default values.
```

Comment:
Is Uid correctly capitalized?  Above,we have many occurrencies of uid.

Reply:
it's all over the shop in the source content. Looks like the keycloak UI uses `Uid` mostly.

Reply:
ok, fine.